# Bioinformatics Spring School 2020


## About

Bioinformatics Spring School 2020 is a week long training event for researchers, supported by [Genomics Aotearoa](https://www.genomics-aotearoa.org.nz) and the [New Zealand eScience Infrastructure (NeSI)](https://www.nesi.org.nz). 

We will combine talks from researchers and hands on computational workshops (influenced by [The Carpentries](https://www.carpentries.org)), covering example workflows of DNA variant calling, genotyping by sequencing (GBS), RNAseq analysis, and environmental DNA (eDNA).



### When and where?

- 23 November - 27 November 2020
- Department of Biochemistry, University of Otago, Dunedin, New Zealand


### Is this for me?

This event is designed to give researchers (including students - 4th year and above) the opportunity to learn about and practice some bioinformatic skills. We'll be covering bioinformatic workflows such as:

- DNA variant calling from next generation sequence data
- Genotype-by-Sequencing analysis (i.e. RAD-Sequencing)
- RNA-sequencing differential expression analysis
- Environmental DNA analysis

along with computational topics such as:

- Introduction to the Unix commandline (Bash)
- Introduction to working in a high performance computing environment (provided by NeSI)
- Introduction to version control using git

This week long event is designed to be an immersive experience so **we ask that you commit to the entire week.** We have chosen to interleave the computational topics throughout the week so that we can build on them throughout the entire week, so this ensures participants have the key knowledge at the right time.

The underlying workshop content is extremely similar to previous workshops run by Genomics Aotearoa so if you've already attended the Genomic Data Carpentry and RNAseq workshops you will have already covered the majority of the content.

## Curriculum


### Bioinformatic topics


#### DNA variant calling from next generation sequence data

- Project organisation
- Assessing read quality
- Trimming and filtering
- Variant calling workflow
- Automating a variant calling workflow


#### Genotype-by-sequencing analysis (RAD-Sequencing)
 
- Data quality assessment
- Calling variant workflow using Stacks (with or without a reference genome)
- Basic population genetic analyses (population structure, phylogenetic tree) 

#### RNA-sequencing differential expression analysis

- Quality assessment
- Trimming and filtering
- Mapping and count
- Differential expression
- Overrepresentation analysis

#### Environmental DNA analysis
 
- Next-generation sequencing approaches for environmental samples (metabarcoding)
- Demultiplexing and quality assessment of NGS amplicon data
- Clustering and denoising of sequence data
- Determine species composition of samples (taxonomic assignment)
- community ecology statistics for eDNA results

### Computational Topics

These topics will be integrated into the bioinformatics workshops and built upon across the week with the goal of demostrating a best practices workflow approach to Bioinformatics analysis.

#### Introduction to the Unix command line (Bash)

Many bioinformatic programs will only operate in a Unix command line environment, as such we need to provide an introduction to working in this environment which will cover:

- Navigating around the filesystem (files/directories/folders)
- Running command line based programs
- Creating scripts to automate workflows


#### Introduction to working in a high performance computing environment

We'll be making use of the New Zealand eScience Infrastructure (NeSI), which is the national provider of high performance computing for researchers, to run our analysis and as part of this we'll cover:

- What/who is NeSI
- Logging onto a remote computer
- Transfer of data to/from a remote computer
- What is a job scheduler and how to use it
- How to request a project allocation on NeSI


#### Introduction to version control using git

Running programs on the command line gives us the opportunity to save the commands used in scripts that we can reuse later on. It's important that as we develop these scripts that we have a machanism to track the changes made. Git is a program that enables us to track the changes we make and see the differences between previous versions. The introduction to version control will cover:

- Setting up Git
- Using git to track changes on scripts
- How to view changes between versions of scripts
- Use git to look at previous versions of scripts


## Apply

This workshop will require participants to provide their own laptop (we will have a limited number of machines to lend)

Expressions of interest are now open. **Please fill in this [form](https://docs.google.com/forms/d/e/1FAIpQLSc-VrWHva2qls4HTR3y1Qm6M1tWXC2vgpx7feo9x87CzHvgHA/viewform?usp=sf_link)**

Applications close 27 October, successful applicants will be notified by the 30th October.